dev/sg/sg_ops.go

```diff
@@ -74,11 +75,18 @@ func opsUpdateImage(ctx context.Context, args []string) error {
 			// We do not want any error handling here, just fallback to anonymous requests
 			writeWarningLinef("Registry credentials are not provided and could not be retrieved from docker config.")
 			writeWarningLinef("You will be using anonymous requests and may be subject to rate limiting by Docker Hub.")
+			dockerCredentials = &credentials.Credentials{ServerURL: "https://index.docker.io/v1/", Username: "", Secret: ""}
 		} else {
 			writeFingerPointingLinef("Using credentials from docker credentials store (learn more https://docs.docker.com/engine/reference/commandline/login/#credentials-store)")
 			opsUpdateImagesContainerRegistryUsernameFlag = &dockerCredentials.Username
 			opsUpdateImagesContainerRegistryPasswordFlag = &dockerCredentials.Secret
 		}
 	}
-	return images.Parse(args[0], *dockerCredentials)
+
+	if *opsUpdateImagesDeploymentKindFlag != "plain" && *opsUpdateImagesDeploymentKindFlag != "helm" {
+		writeFailureLinef("invalid deployment kind: %s", *opsUpdateImagesDeploymentKindFlag)
+		return flag.ErrHelp
+	}
```

Comment:
An easier to read approach with less `&&`:

```suggestion
	switch t := images.DeploymentType(*opsUpdateImagesDeploymentKindFlag) {
	case DeploymentTypeKubernetes, DeploymentTypeHelm:
		return images.Parse(args[0], *dockerCredentials, images.DeploymentType(*opsUpdateImagesDeploymentKindFlag))
	}

	writeFailureLinef("invalid deployment kind: %s", *opsUpdateImagesDeploymentKindFlag)
	return flag.ErrHelp
```